docs/how-to/custom-framing.md

```diff
@@ -0,0 +1,151 @@
+
+1. **Define the Framer and Deframer FPP Components**
+
+   Framer and Deframer components should implement the FPP interfaces by including them.
+
+   In  `MyCustomFramer.fpp`:
+   ```
+    passive component MyCustomFramer {
+        include "path/to/fprime/Svc/Interfaces/FramerInterface.fppi"
+        [...]
+    }
+   ```
+   And in `MyCustomDeframer.fpp`:
+   ```
+    @ Deframer implementation for MyCustomProtocol
+    passive component MyCustomDeframer {
+        include "path/to/fprime/Svc/Interfaces/DeframerInterface.fppi"
+        [...]
+    }
+   ```
+
+2. **Implement the Framer C++ Component**
+
+   Implement the required handler functions:
+   ```cpp
+   // ...existing code...
+   void MyCustomFramer ::dataIn_handler(FwIndexType portNum, Fw::Buffer& data, const ComCfg::FrameContext& context) {
+       // TODO: Implement framing logic
+   }
+
+   void MyCustomFramer ::comStatusIn_handler(FwIndexType portNum, Fw::Success& condition) {
+        this->comStatusOut_out(portNum, condition); // pass comStatus through (unless project requires otherwise)
+   }
+
+   void MyCustomFramer ::dataReturnIn_handler(FwIndexType portNum, Fw::Buffer& data, const ComCfg::FrameContext& context) {
+        // TODO: handle return of data ownership
+        // For example, if component required to allocate from a buffer manager, return the buffer to the manager
+   }
+   // ...existing code...
+   ```
+
+3. **Implement the Deframer C++ Component**
+
+   Similarly, implement the required handler functions:
+   ```cpp
+   // ...existing code...
+   void MyCustomDeframer ::dataIn_handler(FwIndexType portNum, Fw::Buffer& data, const ComCfg::FrameContext& context) {
+       // TODO: Implement deframing logic
+   }
+
+   void MyCustomDeframer ::dataReturnIn_handler(FwIndexType portNum, Fw::Buffer& data, const ComCfg::FrameContext& context) {
+        // TODO: handle return of data ownership
+   }
+   // ...existing code...
+   ```
+
+4. **(Optional) Implement a Frame Detector**
+
+   In some configurations, your communications manager component may produce full frames on reception. This may be the case with some radios that have frame synchronization built-in, or if delimiting frames is handled by another subsystem. 
+   However, if your underlying data transport is stream-based (e.g. TCP, UART) or if you can't guarantee frames will always be received in full by your communications manager, you will need to implement a mechanism to delimit frames. F Prime provides this capability with the [Svc.FrameAccumulator](../../Svc/FrameAccumulator/docs/sdd.md) component, which accumulates a data stream into a circular buffer, and uses a helper `FrameDetector` to detect when a frame is present.
```

Comment:
Phrasing is confusing, starts by saying when it's not needed without being clear about it